### PR TITLE
Add operators for type annotated empty collections

### PR DESCRIPTION
**Pull request by shonfeder**
Broken out of #244

See
https://github.com/informalsystems/apalache/blob/unstable/docs/adr/002adr-types.md#23-dealing-with-bound-variables
for the motivation

Original Author: Igor Konnov <igor@informal.systems>

### CODE REVIEW

File: tlair/src/main/scala/at/forsyte/apalache/tla/lir/oper/TlaFunOper.scala [hunk 1]
```diff
@@ -103,27 +105,29 @@ object TlaFunOper {
...
+    * NameEx("n"),
+    * OperEx(TlaFunOper.app,
+    * OperEx(recFunRef),
+    * OperEx(TlaArithOper.minus, NameEx(n), ValEx(TlaInt(1)))
+    * )
+    * )
+    * ),
+    * NameEx("n"),
+    * TlaIntSet
+    * )
```
Comment (konnov) [inline tlair/src/main/scala/at/forsyte/apalache/tla/lir/oper/TlaFunOper.scala:122]: Shall we exclude this reformatting glitch?

**Review by konnov — APPROVED** [review]
Looks great to me

File: tlair/src/main/scala/at/forsyte/apalache/tla/lir/oper/TlaFunOper.scala (hunk 1 above)
Comment (shonfeder) [reply tlair/src/main/scala/at/forsyte/apalache/tla/lir/oper/TlaFunOper.scala:122]: Yeah, I'd asked about this reformatting on #244 as well, tho this is probably the proper place!

Is the rest of the reformatting in this changeset as intended?

Comment (shonfeder) [reply tlair/src/main/scala/at/forsyte/apalache/tla/lir/oper/TlaFunOper.scala]: I fixed just this block.

**Pull request merged**